common/lib/xmodule/xmodule/js/src/sequence/display.coffee

```diff
@@ -129,12 +157,21 @@ class @Sequence
 
       bookmarked = if @el.find('.active .bookmark-icon').hasClass('bookmarked') then true else false
       @content_container.html(current_tab.text()).attr("aria-labelledby", current_tab.attr("aria-labelledby")).data('bookmarked', bookmarked)
+
+      # update the data-attributes with latest contents only for updated problems.
+      if @anyUpdatedProblems new_position
+        $.each @updatedProblems[new_position], (problem_id, latest_content) =>
+          @content_container
+          .find("[data-problem-id='"+problem_id+"']")
+          .data('content', latest_content)
```

Comment:
@adampalay right now `updatedProblems` structure is like: 
```
{
      seq_pos1: { id1 : "content1", id2: "content2", ... }, 
      seq_pos2: { id3 : "content1", id4: "content2", ... }
      ...
}
```
we are iterating over the updated problems for the given sequence position and update the container with their contents. Later, this container is going to be used by `Problem` module to render the content from data-attr.